_posts/2023-12-11-augmented-reality-in-travel-applications.md

```diff
@@ -0,0 +1,130 @@
+
+## Initialize the camera in the browser
+
+For an executable version, you only need markup - no code (except for the AR.js library). The location-based camera needs a scene to which it must be added to work. It looks like this:
+
+```html
+<a-scene
+  raycaster="objects: [gps-entity-place];"
+  arjs="sourceType: webcam;"
+  embedded
+>
+  <a-camera gps-camera rotation-reader></a-camera>
+</a-scene>
+```
+
+Let's take a look at the attributes:
+
+- `raycaster`: We want to interact with places. To do this, we need to render 3D objects. Under the hood, [AR.js uses Three.js](https://ar-js-org.github.io/AR.js-Docs/ui-events/) for this.
+- `arjs`: Sure, we have to initialize the scene. We also specify `sourceType: webcam` to ensure that the smartphone camera is also used.
+
+For `a-camera` we need `gps-camera` to activate [Location AR](https://ar-js-org.github.io/AR.js-Docs/location-based#camera-component-gps-new-camera-gps-projected-camera-or-gps-camera) and `rotation-reader` to actively track the position when the camera is rotated. This way, we know where the user is pointing their device.
+
+The app was now ready to be tested for the first time via the smartphone browser. To do this, I simply started the server and called up the IP of my machine. Once access to the camera has been granted, a red camera appears in the address bar:
+
+<figure>
+    <img src="img/posts/2023-12-11-augmented-reality-in-travel-applications/iphone-status-bar-while-recording.png" alt="iPhone status bar while recording" class="centered" />
+    <figcaption>The status bar of my iPhone while recording in Safari.</figcaption>
+</figure>
+
+## Interact with remote locations
+
+First of all, we need our own location. To get our current geo-position, we use the [Navigation API](https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API). We now use the callback from `getCurrentPosition` for our further logic:
+
+```js
+Navigator.geolocation.getCurrentPosition((position) => {
+  // the logic
+});
+```
+
+To mark locations visually on the camera, we make use of an `a-box`, which we add to the DOM. With the help of attributes, we can determine its position. We also determine what is rendered. For the first test, I simply opted for blue bars. To inform the scene about the location, we dispatch the event `gps-entity-place-loaded`.
+
+```js
+const icon = document.createElement("a-box");
+icon.setAttribute(
+  "gps-entity-place",
+  `latitude: ${latitude};longitude: ${longitude}`
+);
+icon.setAttribute("name", place.name);
+icon.setAttribute("width", "100");
+icon.setAttribute("height", "35");
+icon.setAttribute("material", "color: #0e55cd");
+
+icon.addEventListener("loaded", () =>
+  window.dispatchEvent(new CustomEvent("gps-entity-place-loaded"))
+);
+```
+
+We have already created a dynamic: The locations are rendered in the camera, during the movement. The only thing missing is interaction.
```

Comment:
What is `dynamic`? Looks like an adjective here. If yes, it requires a following up word.
If it is to be a noun, you should use `dynamics`.